python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -313,11 +332,21 @@ def run(self, inputWarps, skyInfo, **kwargs):
                 compute_n_image=False,
             )
 
+        artifactMasks = kwargs.get("artifactMasks", [None] * len(inputWarps))
+
         # Read in one warp at a time, and accumulate it in all the cells that
         # it completely overlaps.
-        for warpRef in inputWarps:
+        for warpRef, artifactMaskRef in zip(inputWarps, artifactMasks):
             warp = warpRef.get()
 
+            if artifactMaskRef is not None:
+                # Apply the artifact mask to the warp.
+                artifactMask = artifactMaskRef.get()
+                assert artifactMask.getMaskPlaneDict() == warp.mask.getMaskPlaneDict()
+                assert not np.any(warp.mask.array & artifactMask.array)
```

Comment:
Do these statements cover the guards that you mentioned during our discussions. These should all be unnecessary given the global nature of our `Mask`s.

Reply:
Since you reviewed, I realized I have to explicitly add `CLIPPED` and `REJECTED` mask planes again, since they are not native to `warp`. Do you still think these asserts are unnecessary?

Reply:
No, not really.  My thinking was:
 - it'd be more efficient to assert that `getPlaneBitMask(plane)` returns the same thing for all planes of interest;
 - that's a static method so it can't possibly fail;
 - if it's possible for the assert on `getPlaneMaskDict` to fail, then `getPlaneBitMask` would broken;
 - the second assert is possibly meaningful but unlikely to matter, and fairly expensive for something in that category.